STUDIO/org.openl.rules.repository/src/org/openl/rules/repository/db/DBRepository.java

```diff
@@ -352,7 +354,7 @@ public FileData copyHistory(String srcName, FileData destData, String version) t
         Connection connection = null;
         PreparedStatement statement = null;
         try {
-            connection = getConnection();
+            connection = getInitializedConnection();

```

Comment:
Can we put initialization logic to the ``getConnection()`` method?

Reply:
getConnection() is abstract method in DBRepisotory with different implementations in JdbcDBRepository and DatasourceDBRepository.
getConnection() itself is used inside initializeDatabase(), loadDBSettings() and checkRepository().
checkRepository() is used to 1) determine modifications in db if any 2) inside initializeDatabase() to check that repository is already initialized.
These methods are common for all DBRepository implementations.

So if we put initialization logic to the getConnection() method, we should:
1) add DB initialization invocation in each implementation of DBRepository (for every class overriding getConnection() method). If anybody will want to implement a new DBRepository subclass, they must be aware of that requirement.
2) we must properly close Connection inside each getConnection() implementation if initializeDatabase() throws any exception, otherwise we will get connections leak in DB.

So I think that moving initialization logic inside getConnection() method is not a good choice, because it breaks incapsulation, makes each implementation of DBRepository more error-prone and unnecessarily complicates them with almost no profit.

Reply:
If these names are confusing, I can suggest to rename them.
For example current getConnection() rename to createConnection() including in subclasses and getInitializedConnection rename to getConnection()